src/Repl/Repl.tsx

```diff
@@ -0,0 +1,317 @@
+/*
+ * Repl.tsx
+ * Updated Nov. 3rd, 2020
+ * Ben Friedman
+ *
+ * Represents the Read Evaluate Print Loop (Interpreter) of the online BoGL editor
+ */
+
+import React, { useState } from 'react';
+import Terminal from 'terminal-in-react';
+import {apiRequestRunCode} from '../Utilities/BoGLServerRequest';
+import {decodeValue,decodeExprType,decodeError,decodeResponse} from './Decode';
+
+
+// an expression like place(1,b,(1,1)) evaluates to a board and it also adds the same board to
+// the print buffer. This makes sure we do not print duplicate boards.
+// an expression could modify two boards to be identical, in that case it may be confusing to
+// only print one board, but that is a much less common use case that the first one.
+/*
+const extendBoardsString = (board, value, boards) => {
+    let boardStr = decodeBoard(board["value"]);
+    if (boardStr !== value) boards = boards + boardStr + value + "\n";
+}
+*/
+
+
+/*
+ * Repl React function component
+ */
+const Repl = (props) => {
+
+  // Default is an empty buffer
+  const [inputBuffer, setInputBuffer] = useState(Array<any>());
+
+  // input mode: True is input mode, False is regular
+  // default is 'False'
+  const [inputMode, setInputMode] = useState(false);
+
+  // whether to report the type of the current expr
+  // default is 'False'
+  let shouldReportExprType = false;
+
+  // last expression we evaluated in the Repl
+  // when in Input mode, this is the expression we send, and the input buffer
+  // is updated instead
+  const [lastReplExpression, setLastReplExpression] = useState("");
+
+  /*
+   * enterInputMode
+   * Sets input mode true
+   */
+  const enterInputMode = (expr) => {
+    setInputMode(true);
+    setLastReplExpression(expr)
+  }
+
+  /*
+   * exitInputMode
+   * Clears the input buffer, set input mode false, clears replExpression
+   * Used after input has been fully accepted, or 'clear' has been entered
+   */
+  const exitInputMode = () => {
+    setInputBuffer([]);
+    setInputMode(false);
+    setLastReplExpression("")
+  }
+
+  /*
+   * Called when clear is pressed
+   */
+  const clear = () => {
+    exitInputMode();
+    return " 🤖 BoGL Says: Ok, exiting Input Mode. ";
```

Comment:
I wanted to have a check to see if inputMode was false, and tell the user that they are already not reading input, but the value is false already? This is odd, but has something to do with how React does things, however the state as it is for input works nicely w/out adding anything else here.